src/tests/ftest/io/dfuse_sparse_file.py

```diff
@@ -0,0 +1,214 @@
+
+
+class DfuseSparseFile(IorTestBase):
+    """Dfuse Sparse File base class
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DfuseSparseFile object."""
+        super(DfuseSparseFile, self).__init__(*args, **kwargs)
+        self.space_before = None
+        self.ret = None
+
+    def tearDown(self):
+        """Tear down each test case."""
+        try:
+            if self.dfuse:
+                self.dfuse.stop()
+        finally:
+            # Stop the servers and agents
+            super(DfuseSparseFile, self).tearDown()
+
+    def get_remote_file_size(self, file_name):
+        """Obtain remote file size.
+
+          Args:
+            file_name (str): name of remote file
+
+          Returns:
+            integer value of file size
+        """
+
+        cmd = "ssh" " {}@{}" " stat -c%s {}".format(
+            getuser(), self.hostlist_clients[0], file_name)
+        try:
+            result = subprocess.check_output(cmd, shell=True)
+        except subprocess.CalledProcessError as err:
+            self.fail("Get remote file size method failed with: {}".format(err))
+
+        return int(result)
+
+    def get_nvme_free_space(self, display=True):
+        """Display pool free space.
+
+          Args:
+            display (bool): boolean to display output of free space.
+
+          Returns:
+            free_space_nvme (int): Free space available in nvme.
+        """
+        free_space_nvme = self.pool.get_pool_free_space("nvme")
+        if display:
+            self.log.info("Free nvme space: %s", free_space_nvme)
+
+        return free_space_nvme
+
+    def execute_cmd(self, cmd, fail_on_err=True, display_output=True):
+        """Execute cmd using general_utils.pcmd
+
+          Args:
+            cmd (str): String command to be executed
+            fail_on_err (bool): Boolean for whether to fail the test if command
+                                execution returns non zero return code.
+            display_output (bool): Boolean for whether to display output.
+        """
+        try:
+            # execute bash cmds
+            self.ret = general_utils.pcmd(
+                self.hostlist_clients, cmd, verbose=display_output, timeout=300)
+            if 0 not in self.ret:
+                error_hosts = NodeSet(
+                    ",".join(
+                        [str(node_set) for code, node_set in
+                         self.ret.items() if code != 0]))
+                if fail_on_err:
+                    raise CommandFailure(
+                        "Error running '{}' on the following "
+                        "hosts: {}".format(cmd, error_hosts))
+
+        # report error if any command fails
+        except CommandFailure as error:
+            self.log.error("DfuseSparseFile Test Failed: %s",
+                           str(error))
+            self.fail("Test was expected to pass but "
+                      "it failed.\n")
```

Comment:
This feels pretty generic also.  Are you sure this doesn't exist already, and if not, should this be in the general_utils library?